Source/Schedule.hx

```diff
@@ -8,182 +8,195 @@ import haxe.ds.HashMap;
+		cursor.y = yOffset + colHeight * modDay(date);
+		this.addChild(cursor);
+
+		week = -1;
+		makeCols();
+		makeWeekdays();
+		makeTimes();
+		cols.get(today).backgroundColor = 0x55FFFF;
 	}
-	var colorNum = 0;
-	for (lectures in scheduleObject){
-	    for (lecture in lectures){
-		var times = lecture.times.split(" ");
-		var startNums = Util.splitAndParseInt(times[0], "/");
-		//trace("hey");
-		if (startNums[0]<week){
-		    continue;
-		} else if(startNums[0]>week){
-		    break;
+
+	private function modDay(date:Date):Float {
+		return (date.getHours() + (date.getMinutes() + (date.getSeconds()) / 60) / 60) / 24;
+	}
+
+	private function makeCols() {
+		week++;
+
+		for (j in 0...7) {
+			var col:TextField = new TextField();
+			addChild(col);
+
+			col.x = xOffset + j * colWidth;
+			col.y = yOffset;
+			col.text = Std.string(endDate.getDate());
+
+			col.background = true;
+			col.backgroundColor = 0xFFFFFF;
+			col.border = true;
+			col.borderColor = 0x000000;
+			col.width = colWidth;
+			col.height = colHeight;
+
+			cols.set(Day.fromDate(endDate), col);
+			endDate = DateTools.delta(endDate, DateTools.days(1));
+		}
+
+		var sunday = DateTools.delta(endDate, -DateTools.days(7));
+		for (lecture in scheduleObject) {
+			var lecCursor:Int = lecture.curLectureNum;
+			do {
+				// If lecture is not held on that day continue to next lecture
+				if (lecture.exceptions.indexOf(lecCursor) != -1) {
+					lecCursor++;
+					if (lecture.weekdays.length == 1) break;
+					continue;
+				}
+				var index = lecCursor % lecture.weekdays.length;
+				var weekday = lecture.weekdays[index];
+				var start = lecture.times[index].start;
+				var date = DateTools.delta(sunday, DateTools.days(weekday) + DateTools.hours(start[0]) + DateTools.minutes(start[1]));
+
+				// If lecture is not held within the range of this week
+				if (!Util.DayinRange(lecture.startDate, lecture.endDate, date)){
+					lecCursor++;
+					if (lecture.weekdays.length == 1) break;
+					continue;
+				}
+
+				// Lecture will be added since it is on this day in range
+
+				var end = lecture.times[index].end;
+				var startFrac = (start[0] + start[1] / 60) / 24;
+				var yPos = yOffset + colHeight * startFrac;
+				var xPos = xOffset + weekday * colWidth;
+
+				var endFrac = (end[0] + end[1] / 60) / 24;
+				var h = colHeight * (endFrac - startFrac);
+				var lectureCol = new TextField();
+				addChild(lectureCol);
+
+				lectureCol.x = xPos;
+				lectureCol.y = yPos;
+				lectureCol.text = lecture.title;
+
+				lectureCol.background = true;
+				lectureCol.backgroundColor = colors[scheduleObject.indexOf(lecture)];
+				lectureCol.border = true;
+				lectureCol.borderColor = 0x000000;
+				lectureCol.width = colWidth;
+				lectureCol.height = h;
+				lecCursor++;
+
+				lectureCols.push(lectureCol);
+				if (lecture.weekdays.length == 1) break;
+			} while (lecture.weekdays[lecCursor % lecture.weekdays.length] >= lecture.weekdays[(lecCursor - 1) % lecture.weekdays.length]);
```

Comment:
A better condition is `lecCursor % lecture.weekdays.length == 0`.

Reply:
I think a similar thing can be done as I commented above. That is check if delta between this lecture and the next one is < 0. This will additionally allow cycles such as these [1,3,4,4] where the 2nd 4 occurs during a time before the 1st 4. This however doesn't allow two week cycles of the type ```weekdays = [2,2], times = [10:30-11:30, 14:30-15:30]``` which is also currently not capable.

Reply:
That's a good point that such cycles are not capable.  Maybe my idea of doing two week cycles like `weekdays = [1, 3, 5, 2, 4]` is shortsighted because it doesn't allow for periods of more than 7 days between consecutive classes.  What if we used `weekdays = [1, 3, 5, 9, 11]` instead and also had a field `cycleLength = 14`?